frontend/pages/create-project.tsx

```diff
@@ -0,0 +1,221 @@
+import React, {useState, useEffect} from 'react';
+import Link from 'next/link';
+import { PhotoIcon } from '@heroicons/react/24/solid';
+import { useFormik } from 'formik';
+
+interface CreateProjectProps {}
+
+const CreateProject: React.FC<CreateProjectProps> = () => {
+    const maxCharacterLimit = 20;
+    const [materialsList, setMaterialsList] = useState<string[]>([]);
+
+    const formik = useFormik({
+        initialValues: {
+            projectName: '',
+            aboutProject: '',
+            selectedFile: null,
+            listOfParticipant: [] as string[],
+            newList: '',
+            material: [] as string[],
+            newMaterial: '',
+            error: '',
+        },
+        onSubmit: () => {
+            // Handle form submission
+        },
+    });
+
+    const handleAboutProjectChange = (e: React.ChangeEvent<HTMLTextAreaElement>) => {
+        const inputValue = e.target.value;
+        if (inputValue.length <= maxCharacterLimit) {
+            formik.setFieldValue('aboutProject', inputValue);
+        }
+    };
+
+    const handleFileChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+        const file = e.currentTarget.files?.[0];
+        formik.setFieldValue('selectedFile', file);
+    };
+
+    const addParticipant = () => {
+        if (formik.values.newList.trim() !== '') {
+            formik.setFieldValue('listOfParticipant', [...formik.values.listOfParticipant, formik.values.newList]);
+            formik.setFieldValue('newList', '');
+        }
+    };
+
+    const removeParticipant = (index: number) => {
+        formik.setFieldValue('listOfParticipant', formik.values.listOfParticipant.filter((_, i) => i !== index));
+    };
+
+    const isMaterialValid = (material: string) => {
+        return materialsList.includes(material);
+    };
+
+    const addMaterial = () => {
+        if (formik.values.newMaterial.trim() !== '' && isMaterialValid(formik.values.newMaterial)) {
+            formik.setFieldValue('material', [...formik.values.material, formik.values.newMaterial]);
+            formik.setFieldValue('newMaterial', '');
+            formik.setFieldValue('error', '');
+        } else {
+            formik.setFieldValue('error', 'This material is not available');
+        }
+    };
```

Comment:
Autocompletion on elements could be nicer (you don't allow student to use non existing materials)
In that case, this code will become legacy